src/main/java/it/gov/pagopa/payment/service/PDVServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package it.gov.pagopa.payment.service;
+
+import it.gov.pagopa.payment.connector.decrypt.DecryptRestConnector;
+import it.gov.pagopa.payment.connector.encrypt.EncryptRestConnector;
+import it.gov.pagopa.payment.dto.CFDTO;
+import it.gov.pagopa.payment.exception.custom.PDVInvocationException;
+import org.springframework.stereotype.Service;
+
+import java.util.function.Supplier;
+
+@Service
+public class PDVServiceImpl implements PDVService {
+
+    private final DecryptRestConnector decryptRestConnector;
+    private final EncryptRestConnector encryptRestConnector;
+
+    public PDVServiceImpl(DecryptRestConnector decryptRestConnector, EncryptRestConnector encryptRestConnector) {
+        this.decryptRestConnector = decryptRestConnector;
+        this.encryptRestConnector = encryptRestConnector;
+    }
+
+    @Override
+    public String encryptCF(String fiscalCode) {
+        return wrapPDVCall(() -> encryptRestConnector.upsertToken(new CFDTO(fiscalCode)).getToken(),
+                "An error occurred during encryption");
+    }
+
+    @Override
+    public String decryptCF(String userId) {
+        return wrapPDVCall(() -> decryptRestConnector.getPiiByToken(userId).getPii(),
+                "An error occurred during decryption");
+    }
+
+    private <T> T wrapPDVCall(Supplier<T> action, String errorMessage) {
```

Comment:
do we really needs the use of generic? It looks like the method is always called with String type lambdas. Simplify if we can, thanks!